src/sgame/sg_bot_skilltree.cpp

```diff
@@ -218,14 +243,22 @@ static bool SkillIsAvailable(const botSkillTreeElement_t &skill, team_t team, sk
 }
 
 // Note: this function modifies possible_choices to remove the one we just chose.
-static Util::optional<botSkillTreeElement_t> ChooseOneSkill(team_t team, skillSet_t skillSet, std::vector<botSkillTreeElement_t> &possible_choices, std::mt19937_64& rng)
+static Util::optional<botSkillTreeElement_t> ChooseOneSkill(team_t team, skillSet_t skillSet, std::vector<botSkillTreeElement_t> &possible_choices, std::mt19937_64& rng, int skillLevel)
```

Comment:
Nit: I think the algorithm would be cleaner if there was a separate function that collects all the base skills and ChooseOneSkill only chooses random ones

Reply:
That makes sense. I think I'll keep it like this because it's better tested though.